### DISCUSSION TITLE
PackageBuildOutputs.groovy is creating and deploying the package in reopsitory/null/default/default.tar

### DISCUSSION THREAD

**Discussion by ratnamGT** [IBM/dbb#259] [Q&A]
I am creating a packaging and deploying it to jfrog Artifactory.
below is the build script
$DBB_HOME/bin/groovyz  var/dbb/Pipeline/PackageBuildOutputs/PackageBuildOutputs.groovy \
                      --workDir /u/myproject/project/BUILD \
                      -p \
                      -aprop appArtifactRepository.properties \
                      -au http://jfrogurl/artifactory \
                      -ar example-repo-local/ \
                      --tarFileName default.tar  \
                      -aU user \
                      -aP ******* \
                      --verbose
echo "packaging finished"   

It is creating package default.tar and uploading it to the http://Jfrog/artifactory/example-repo-local/null/Default/package.tar .It is unnecessary creating null/Default folder inside my example-repo-local . I tried to put -ad <directory path> but it is failing with error that this option for wazi deploy.I am not using wazi deploy and I do not want to deploy package in null/Default . I want to deploy my defalut.tar file in my artifactory example-repo-local directly . How can I avoid this to create null/Default inside my repo.
It looks like ,it is appending null/Default in below groovy script.
![image](https://github.com/IBM/dbb/assets/172598523/d6000b9d-940e-429d-b1cb-5a4d42623c8b)

**Comment by dennis-behm** [comment]
Hi @ratnamGT, 

The `appArtifactRepository.properties` manages the directory within the artefact repo. 
https://github.com/IBM/dbb/blob/main/Pipeline/PackageBuildOutputs/appArtifactRepository.properties#L18-L19

Alternatively you can define it via the CLI parm `-ad,--artifactRepositoryDirectory` if you don't plan to have an `appArtifactRepository.properties` within the application repository.

This will then put it into the correct folder.

**Reply by dennis-behm** [reply]
It actually looks like we are missing to make the directory mandatary if you publish it via the ArtifactRepoHelper ...

**Comment by ratnamGT** [comment]
Thanks @dennis-behm I will try to change as per your suggestion
